go/libkb/saltpack_enc.go

```diff
@@ -18,8 +17,13 @@ func SaltPackEncrypt(
 	for _, k := range receivers {
 		receiverBoxKeys = append(receiverBoxKeys, naclBoxPublicKey(k))
 	}
-	plainsink, err := saltpack.NewEncryptArmor62Stream(
-		sink, naclBoxSecretKey(sender), receiverBoxKeys)
+
+	var bsk saltpack.BoxSecretKey
+	if !sender.IsNil() {
```

Comment:
The `sender` here has a 0-NaClPublic key, which is zero-ish.  I guess alternatively we can check that `Private` is actually the `nil` pointer, I hadn't thought of that...